### DISCUSSION TITLE
gitignore generated files?

### DISCUSSION THREAD

**Discussion by gvanderclay** [ivanhofer/typesafe-i18n#500] [Q&A]
Hello!
First off I want to say I appreciate your work on this library. I've been trying to use i18next but the lack of type safety drove me to this library. Now to my question.

Is it required to commit the generated files? I usually prefer `.gitignore` generated files since it reduces the likelihood of people changing them without changing the data they are derived from.

Thanks!

**Comment by ivanhofer** [comment]
Hi @gvanderclay thanks for your kind words :)

I personally commit the file. I think it is also the recommended way of other type generators like e.g. for GraphQl.
If you would add the file (or all files that are beeing generated), you would probably need to add a `postinstall` script that runs the generator, or else you would have invalid code when you or a teammate checks out the repository the first time. The same applies to a CI process.
